Caltech-IPAC/firefly
# Firefly-51: Make FieldGroupConnector use function components with hooks

Firefly-51: Make FieldGroupConnector use function components with hooks

  - Created a new field group connector hook
  - converted all `FieldGroup` related components to be functions with hooks
  - updated and cleaned up `NaifidPanel`, it needed a little work
  - updated many propTypes to include shape of initialState
  - removed value from some components props, replaced with initialState:{value}
  - a attempt to better define what should be in initial state
  - hook give a warning is value is passed as prop

_Code to look at_:
 
  - Look at `FieldGroupConnect.jsx`, look at the hook `useFieldGroupConnector`. Don't bother looking at the differences, just look at the new file.
  - All other files are pretty straight forward. You might want to look at a couple of files or focus on the components that you wrote. 
 - Before all code used the HOC `fieldGroupConnector`, now they use the hook `useFieldGroupConnector`. Notice how much easier it is follow how to setup the hook compared to the HOC.
 - @deannaji look at NaifidPanel.jsx.  I had to make some changes, we should discuss it soon.

_to test:_
https://irsawebdev9.ipac.caltech.edu/firefly-51-field-group-with-hooks/firefly/
https://irsawebdev9.ipac.caltech.edu/with-hooks/irsaviewer/

- Check console while testing, look for errors 
- Mostly everything should work as before
 - Critical Stuff:
    - Gator Catalog Search
    - Tap Search
    - Image Search Panel
    - workspace connection
    - naif resolving
